gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/api/use_case/UpdateDynamicPropertiesUseCase.java

```diff
@@ -103,11 +103,10 @@ public void execute(Input input) {
         );
 
         final boolean isApiSynchronized = apiStateDomainService.isSynchronized(updated, auditInfo);
-
         // Deploy only if
         // - API is not synchronized: no manual changes
         // - properties have changed
-        if (isApiSynchronized && needRedployment(api.getApiDefinitionHttpV4().getProperties(), previousProperties)) {
+        if (!isApiSynchronized && needRedployment(api.getApiDefinitionHttpV4().getProperties(), previousProperties)) {
```

Comment:
This comment is false. You should not trust it.
See my main comment for explanation

Reply:
No. Because if I'm currently working on my API in the policy studio, I don't want my current work to be deployed, even if properties have been automatically updated.

I agree that there is a bug, but the solution is not to force a deployment.